src/main/java/eu/darkbot/api/items/ItemUseResult.java

```diff
@@ -0,0 +1,65 @@
+package eu.darkbot.api.items;
+
+import eu.darkbot.api.managers.HeroItemsAPI;
+
+import java.util.function.Consumer;
+
+/**
+ * {@link HeroItemsAPI#useItem(SelectableItem, ItemUseFlag...)} results
+ */
+public enum ItemUseResult {
+    /**
+     * Item was used successfully
+     */
+    SUCCESS,
+    /**
+     * Failed to use an item
+     */
+    FAILED,
+    /**
+     * Item isn't ready
+     *
+     * @see Item#isReady()
+     * @see ItemUseFlag#READY
+     */
+    NOT_READY,
+    /**
+     * Item isn't available to use by either implementation and/or in-game.
+     */
+    NOT_AVAILABLE,
+    /**
+     * Item is already selected if {@link ItemUseFlag#NOT_SELECTED} flag was used
+     *
+     * @see ItemUseFlag#NOT_SELECTED
+     */
+    ALREADY_SELECTED,
+    /**
+     * Item have insufficient quantity, non-positive, <= 0
+     * Only if {@link ItemUseFlag#POSITIVE_QUANTITY} flag was used
+     *
+     * @see ItemUseFlag#POSITIVE_QUANTITY
+     */
+    INSUFFICIENT_QUANTITY;
+
+    /**
+     * Is executed only on {@link #SUCCESS} result
```

Comment:
This is too strict and limiting definition of the API, if in the future we added other successful results we wouldd be forced to count them as ifFailed not to break the contract.

Say that something along the lines of: `It's executed on successful using of the item, like {@link #SUCCESS}` that way you don't close off the posibility of more than just success being there.

Also maybe consider adding `isSuccessful` that currently returns `this == SUCCESS` but in the future could be expanded. You can put this explanation on `isSuccessful` directly and make ifSuccessful and ifFailed reference it